src/main/java/com/mecatran/gtfsvtor/validation/dao/DropOffPickUpTypeForSplitOrJoinedTripsValidator.java

```diff
@@ -0,0 +1,179 @@
+import com.mecatran.gtfsvtor.reporting.ReportSink;
+import com.mecatran.gtfsvtor.reporting.issues.WrongDropOffPickUpTypeForSplitTripsIssue;
+import com.mecatran.gtfsvtor.validation.DaoValidator;
+
+public class DropOffPickUpTypeForSplitOrJoinedTripsValidator
+		implements DaoValidator {
+
+	/**
+	 * Validates that boarding restrictions for split and joined trains are
+	 * correctly set.
+	 * Boarding restrictions (pickup / drop off type) are set correctly, if
+	 * for the stops in common only one of these trips allows pickup (in case
+	 * of joined trains) or drop-off (in case of split trains).
+	 * <p>
+	 * Note that this validation assumes, that split/joined trips have the same service id.
+	 *
+	 * @param context validation context
+	 */
+	@Override
+	public void validate(Context context) {
+		IndexedReadOnlyDao dao = context.getDao();
+		ReportSink reportSink = context.getReportSink();
+
+		// Process trips route by route to reduce memory usage
+		// split trains have at least their two first stops in common, joined
+		// at least their last stops. We assume that both trips have the same
+		// service id. That means stop id, departure and arrival time
+		// match so we collect all split/joined trips by a key representing
+		// their first/last to stops and service id.
+		dao.getRoutes().forEach(route -> {
+			// TODO validation could be restricted to trainish (or configurable) route types.
+			ListMultimap<Object, GtfsTrip.Id> tripsPerStartKey = ArrayListMultimap.create();
+			ListMultimap<Object, GtfsTrip.Id> tripsPerEndKey = ArrayListMultimap.create();
+			dao.getTripsOfRoute(route.getId()).forEach(trip -> {
+				List<GtfsStopTime> stopTimesOfTrip = dao.getStopTimesOfTrip(
+						trip.getId());
+				Object tripStartKey = computeTripKey(trip, stopTimesOfTrip, true);
+				tripsPerStartKey.put(tripStartKey, trip.getId());
+
+				Object tripEndKey = computeTripKey(trip, stopTimesOfTrip, false);
+				tripsPerEndKey.put(tripEndKey, trip.getId());
+
+			});
+
+			// validate that of split trains, at least one has dop-off none set
+			// for stops in common
+			for (List<GtfsTrip.Id> splitTripsIds : Multimaps.asMap(
+					tripsPerStartKey).values()) {
+				/*
+				 * This loop is O(n^2), but we expect to be usually max 2 trips with
+				 * same start stops at the same time.
+				 */
+				for (int i = 0; i < splitTripsIds.size(); i++) {
+					for (int j = i + 1; j < splitTripsIds.size(); j++) {
+						GtfsTrip.Id tripId1 = splitTripsIds.get(i);
+						GtfsTrip.Id tripId2 = splitTripsIds.get(j);
+						if (haveBothDropOffInSameStop(dao.getStopTimesOfTrip(tripId1),
+								dao.getStopTimesOfTrip(tripId2))) {
+							reportSink.report(
+									new WrongDropOffPickUpTypeForSplitTripsIssue(tripId1,
+											tripId2, true));
+						}
+					}
+				}
+			}
+			// validate that of split trains, at least one has dop-off none set
+			// for stops in common
+			for (List<GtfsTrip.Id> joinedTripsIds : Multimaps.asMap(
```

Comment:
Should not we test also if trip headsigns for joined trains are the same?
> Set each section of a joined train to display the same destination on the trip_headsign.

Reply:
Yes, I thought about this, too. Check is added.